lib/services/magic.ts

```diff
@@ -0,0 +1,59 @@
+import sgMail from '@sendgrid/mail';
+import { createToken, hashToken } from '../helpers/backend/tokens';
+import { prisma } from './db';
+import { createSession } from './session';
+import { findOrCreateUser } from './user';
+
+sgMail.setApiKey(process.env.SENDGRID_API_KEY!);
```

Comment:
Is the `!` intentional? I might misunderstand, but doesn't this just turn it to `false`?

Reply:
This is just me being lazy. Since `process.env.SENDGRID_API_KEY` is of type `string | undefined`, adding the `!` unsafely narrows it to `string`. I will add error handling.